xgi/communities/spectral.py

```diff
@@ -43,34 +41,48 @@ def spectral_clustering(H, k=None):
         Advances in Neural Information Processing Systems.
 
     """
-    if k is None:
-        raise NotImplementedError(
-            "Choosing a number of clusters organically is currently unsupported. Please specify an integer value for paramater 'k'!"
+    if k > H.num_nodes:
+        raise XGIError(
+            "The number of desired clusters cannot exceed the number of nodes!"
         )
-    else:
-        if k > H.num_nodes:
-            raise XGIError(
-                "The number of desired clusters cannot exceed the number of nodes!"
-            )
 
     # Compute normalize Laplacian and its spectra
     L, rowdict = normalized_hypergraph_laplacian(H, index=True)
     evals, eigs = eigsh(L, k=k, which="SA")
 
     # Form metric space representation
     X = np.array(eigs)
-    print(X.shape, X)
 
     # Apply k-means clustering
-    _clusters = _kmeans(X, k)
+    _clusters = _kmeans(X, k, **kwargs)
 
     # Remap to node ids
     clusters = {rowdict[id]: cluster for id, cluster in _clusters.items()}
 
     return clusters
 
 
-def _kmeans(X, k, seed=37):
+def _kmeans(X, k, max_iter=1_000, seed=None):
+    """Simple k-means clustering of vectors X.
+
+    Uses Forgy method for selecting initial centroids.
+
+    Parameters
+    ----------
+    X : array-like
+        Vectors to cluster.
```

Comment:
specify shape to avoid ambiguity, i guess it's (n,d) for n vectors in d dimensions?